test/integration/simulacron-helper.js

```diff
@@ -0,0 +1,298 @@
+
+var simulacronHelper = {
+  _execute: function(processName, params, cb) {
+    var originalProcessName = processName;
+
+    // If process hasn't completed in 10 seconds.
+    var timeout = undefined;
+    if(cb) {
+      timeout = setTimeout(function() {
+        cb("Timed out while waiting for " + processName + " to complete.");
+      }, 10000);
+    }
+
+    var p = spawn(processName, params, {});
+    p.stdout.setEncoding('utf8');
+    p.stderr.setEncoding('utf8');
+    p.stdout.on('data', function (data) {
+      console.log("%s_out> %s", originalProcessName, data);
+    });
+
+    p.stderr.on('data', function (data) {
+      console.log("%s_err> %s", originalProcessName, data);
+    });
+
+    p.on('close', function (code) {
+      console.log("%s exited with code %d", originalProcessName, code);
+      if(cb) {
+        clearTimeout(timeout);
+        if (code === 0) {
+          cb();
+        } else {
+          cb(Error("Process exited with non-zero exit code: " + code));
+        }
+      }
+    });
+
+    return p;
+  },
+  start: function(cb) {
+    var self = this;
+    var simulacronJarPath = process.env['SIMULACRON_PATH'];
+    if (simulacronJarPath == undefined) {
+      throw new Error("SIMULACRON_PATH system environment not set");
+    }
+    var processName = 'java';
+    var params = ['-jar', simulacronJarPath, '--ip', '127.0.0.101'];
+    var initialized = false;
+
+    var timeout = setTimeout(function() {
+      cb(new Error("Timed out while waiting for ADS server to start."));
+    }, 10000);
+
+    self.sProcess = self._execute(processName, params, function() {
+      if(!initialized) {
+        cb();
+      }
+    });
+    self.sProcess.stdout.on('data', function (data) {
+      // This is a bit of a kludge, check for a particular log statement which indicates
+      // that all principals have been created before invoking the completion callback.
+      if(data.indexOf('Started HTTP server interface') !== -1) {
+        clearTimeout(timeout);
+
+        helper.trace('Simulacron initialized!');
+        initialized = true;
+        cb();
+      }
+    });
+  },
+  stop: function(cb) {
+    if(this.sProcess !== undefined) {
+      if(this.sProcess.exitCode) {
+        console.log("Server already stopped with exit code %d.", this.process.exitCode);
+        cb();
+      } else {
+        this.sProcess.on('close', function () {
+          cb();
+        });
+        this.sProcess.on('error', cb);
+        if (process.platform.indexOf('win') === 0) {
+          var params = ['Stop-Process', this.sProcess.pid];
+          this._execute('powershell', params, cb);
+        } else {
+          this.sProcess.kill('SIGINT');
+        }
+
+      }
+    } else {
+      cb(Error("Process is not defined."));
+    }
+  },
+  baseAddress: "http://localhost:8187",
+  SimulacronCluster: SimulacronCluster
+};
+
+function SimulacronCluster() {
+  this.baseAddress = simulacronHelper.baseAddress;
```

Comment:
We may get to point where we want to configure ip address ranges and http interface address, but I think it's fine to make it not configurable as you have for now.